src/components/kpi/kpi.directive.coffee

```diff
@@ -36,11 +36,18 @@ angular
             if kpiTemplate? && kpiTemplate.extra_params?
               $scope.kpi.possibleExtraParams = kpiTemplate.extra_params
 
-            $scope.kpi.targets ||= []
-            if !_.isEmpty($scope.kpi.targets[0])
+            $scope.kpi.data ||= {}
+            $scope.kpi.results ||= {}
+            $scope.kpi.targets ||= {}
+
+            $scope.watchableData = $scope.kpi.data[$scope.kpi.element_watched]
```

Comment:
have you tested that when you edit the settings? because on mine it's broken (wouldn't it be better to keep reading directly the kpi object?).
I think that's because when you update the settings, it calls UPDATE, which calls SHOW, which does not update $scope.watchableData